### PR TITLE
balochi_phonetic - keys rearranged, some added, mobile added

### PR DESCRIPTION
**Pull request by rmlockwood**

### CODE REVIEW

File: balochi_phonetic/source/balochi_phonetic.kps [hunk 1]
```diff
@@ -1,2 +1,2 @@
...
...&amp;Install</Caption><URL></URL><Left>0</Left><Top>0</Top></Button><Button><ButtonType>cancel</ButtonType><Caption>Cancel</Caption><URL></URL><Left>0</Left><Top>0</Top></Button><Button><ButtonType>about</ButtonType><Caption>&amp;About...</Caption><URL></URL><Left>0</Left><Top>0</Top></Button></Buttons><Registry/><StartMenu><Folder></Folder><Items/></StartMenu><Info><Version URL="">1.1</Version><Name URL="">Balochi Phonetic</Name><Copyright URL="">© 2017 SIL International. This keyboard is distributed under The MIT License (MIT).</Copyright></Info><Files><File><Name>Balochi_Phonetic.kvk</Name><Description>File Balochi_Phonetic.kvk</Description><CopyLocation>0</CopyLocation><FileType>.kvk</FileType></File><File><Name>Balochi_Phonetic.ico</Name><Description>File Balochi_Phonetic.ico</Description><CopyLocation>0</CopyLocation><FileType>.ico</FileType></File><File><Name>welcome\welcome.htm</Name><Description>File welcome.htm</Description><CopyLocation>0</CopyLocation><FileType>.htm</FileType></File><File><Name>..\build\balochi_phonetic.kmx</Name><Description>Keyboard Balochi Phonetic</Description><CopyLocation>0</CopyLocation><FileType>.kmx</FileType></File></Files><Strings/></Package>
```
Comment (mcdurdin) [inline balochi_phonetic/source/balochi_phonetic.kps]: The file balochi_phonetic.kvk should be a reference to the version in the ../build/ folder, not the one in the source folder.

**Comment by rmlockwood** [comment]
Ok changed the kps file and the kpj file. Ron

**Comment by DavidLRowe** [comment]
- [ ] The welcome.htm doesn't display correctly. 
Source has:
image
Display is:
image
Inserting `<meta content="charset=utf-8" />` after the `<head>` element seems to fix this.

- [ ] On the touch layout (testing on iphone 4s) when the shift key on the default layer is pressed it turns into a blank key. On other keyboard layouts, it changes background color, but still has the arrow.

- [ ] On the numeric layer of the touch layout, the ten digits keys need a long press to access the Arabic style digit, but due to a bug in Keyman, currently long press keys don't work. @mcdurdin Is this likely to be fixed soon during the current sprint? I'm not finding it in the list of issues.

**Comment by mcdurdin** [comment]
@DavidLRowe where were you testing the longpress issue? The longpress issue was corrected in KeymanWeb 2.0.473 which is also embedded in Keyman for Android 2.8.300, Keyman for iPhone and iPad 2.6.4, and in the latest Keyman Developer 10 alpha release. It has not been patched in Keyman Developer 9.

Note, for correctness, the charset declaration needs to be either  `<meta charset="utf-8">` or `<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />`.

**Comment by rmlockwood** [comment]
Should I make the fix for the first item? I don’t know how to fix the 2nd item. 3rd item – does it apply?

Thanks,

Ron

**Comment by DavidLRowe** [comment]
@rmlockwood Yes, inserting the `<meta>` tag should be done to fix the first item. 

For the second item, I think it's usual to include `"text": "*Shift*"` (where currently there is `"text": ""`) on the two K_SHIFT entries which have `"nextlayer": "default"`. This displays the shift arrow on the key to give the user an indication how to get out of shift mode.

Don't worry about the third item; that bug has been fixed. 

@mcdurdin Using Chrome on an old iPhone 4s for testing, connecting to Keyman Developer via 192.168.1.8:8008. (Long press in the Keyman app itself works.) The Keyman app says it's 2.6.4, but from the iPhone home screen, Settings, Keyman says 2.5.2. After using App Store, update and restarting the iPhone, Keyman no longer shows up in Settings, Keyman, however the longpress doesn't work from Chrome connecting to Keyman Developer. Very weird, but my problem, not this keyboard.

**Comment by rmlockwood** [comment]
Done with those two fixes.

 

Ron

**Comment by DavidLRowe** [comment]
The fixes are in, but unfortunately two .bak files slipped in as well. Please delete `source/welcome/welcome.htm.bak` and `source/balochi_phonetic-layout.js.bak`. Thanks!

**Comment by rmlockwood** [comment]
Ok, I deleted them through the github web interface. Hopefully that will work. Thanks, Ron

**Comment by DavidLRowe** [comment]
The `<meta content="charset="utf-8" />` statement in the `welcome.htm` file is incorrect. Please change to `<meta charset="utf-8" />`.

**Comment by rmlockwood** [comment]
done

**Comment by DavidLRowe** [comment]
This PR should be ready to merge.

**Comment by rmlockwood** [comment]
I could really use that new version of the balochi_phonetic keyboard in the next 12 hours if it is at all possible. We are at a workshop where we need it.

 

Thanks,

Ron

**Comment by mcdurdin** [comment]
I've published the keyboard to http://keymankeyboards.com/?id=718 for Windows and macOS. However, the build for iOS and Android will take a little longer as you caught us in middle of the process of upgrading the server which builds and deploys those files.

**Comment by mcdurdin** [comment]
Okay, iOS and Android downloads should now be available.

**Comment by rmlockwood** [comment]
Thanks so much!

**Pull request merged**